### ISSUE TITLE
Improve hot reload speed and process to enable fast development-test cycle

### ISSUE THREAD

**Issue by MarcSkovMadsen** [bokeh/bokeh#9501]
THIS ISSUE WAS FIRST FILED TO PANEL HERE https://github.com/holoviz/panel/issues/849

BUT I WAS INFORMED ITS A FEATURE REQUEST FOR BOKEH

#### System Info

- Panel: 0.7.0
- Bokeh: 1.4.0
- Python: 3.7.4
- OS: Windows 8.1
- Browser: Chrome

#### My Pain

When developing apps in Panel if often wan't to see how the change looks. It's possible to do by using the `--dev` flag but

- The restart of the Panel/ Bokeh server is very slow compared to the alternative technologies of Streamlit and Angular that I'm used to.
- The browser window does not automatically reload. I have to wait for the bokeh server to reload and then I can click refresh. In Streamlit and Angular I do not have to do this. I don't have to do this when I work with Sphinx either.

![hot_reload_process_slow](https://user-images.githubusercontent.com/42288570/70203790-68a01200-171e-11ea-9d20-400bd9fed99f.gif)

(Note I have to run panel server as a *python module* in order for reload to work on windows).

#### Solutions

- Improve restart time for Panel/ Bokeh server
- Enable automatic reload of the browser.

I believe enabling automatic reload of the browser alone would be very, very valuable if its difficult to improve the restart time for the Panel/ Bokeh server.

**Comment by bryevdv** [comment]
@MarcSkovMadsen Are you interested to work on this?

**Comment by MarcSkovMadsen** [comment]
@bryevdv . I would not be able to find the time. My focus is on streamlit and panel for now.

**Comment by bryevdv** [comment]
well I certainly want to make opportunities and space for new contributors, for sure, but in this case it's also a matter of very limited bandwidth on our end. With all the other 2.0 work ongoing I doubt this will be able to be prioritized until at least a few months in to 2020 or later (just setting expectations).

**Comment by MarcSkovMadsen** [comment]
@bryevdv . I appreciate your feedback and I understand. Thanks.

**Comment by MarcSkovMadsen** [comment]
Working on a workaround for this (and other things) in Panel here https://discourse.holoviz.org/t/awesome-panel-designer/643/3 to speed up my development experience.

**Comment by bryevdv** [comment]
@philippjfr @MarcSkovMadsen were there every any developments on the panel side of things relevant to this? i.e. concrete ideas that precipitated or PRs that could be ported over?

**Comment by MarcSkovMadsen** [comment]
Yes. Panel is now using `watchfiles` to provide hot reload. It is a great experience.
